src/razor/src/razorLanguageServerOptionsResolver.ts

```diff
@@ -35,23 +35,19 @@ export function resolveRazorLanguageServerOptions(
 }
 
 function findLanguageServerExecutable(withinDir: string) {
-    const extension = isWindows() ? '.exe' : '';
-    const executablePath = path.join(withinDir, `rzls${extension}`);
+    const isSelfContained = fs.existsSync(path.join(withinDir, 'coreclr.dll'));
     let fullPath = '';
-
-    if (fs.existsSync(executablePath)) {
-        fullPath = executablePath;
+    if (isSelfContained) {
+        const fileName = isWindows() ? 'rzls.exe' : 'rzls';
+        fullPath = path.join(withinDir, fileName);
     } else {
-        // Exe doesn't exist.
-        const dllPath = path.join(withinDir, 'rzls.dll');
-
-        if (!fs.existsSync(dllPath)) {
-            throw new Error(
-                vscode.l10n.t("Could not find Razor Language Server executable within directory '{0}'", withinDir)
-            );
-        }
+        fullPath = path.join(withinDir, 'rzls.dll');
+    }
 
-        fullPath = dllPath;
+    if (!fs.existsSync(fullPath)) {
+        throw new Error(
+            vscode.l10n.t("Could not find Razor Language Server executable within directory '{0}'", withinDir)
```

Comment:
I know the old code didn't do this either, but while you're here anyway, it occurs to me it would be much more helpful if this logged `fullPath`, so the user would also know which file it was looking for. I've hit this message before and not been sure if it was the DLL or EXE that was an issue.